src/unix/tcp.c

```diff
@@ -359,6 +370,22 @@ int uv_tcp_nodelay(uv_tcp_t* handle, int on) {
 }
 
 
+int uv_tcp_fastopen(uv_tcp_t* handle, int qlen) {
+  int err;
+
+  if (uv__stream_fd(handle) != -1) {
+    err = uv__tcp_fastopen(uv__stream_fd(handle), qlen);
+    if (err)
+      return err;
+  }
+
+  if (qlen)
```

Comment:
Let's not do this. If there is no socket, fail with EBADF. This is already troublesome with the keepalive options and I wouldn't want to make the same mistake.